modules/user/user.model.js

```diff
@@ -65,4 +72,19 @@ const userSchema = new mongoose.Schema({
   confirmationToken: String,
 });
 
+userSchema.pre('findOneAndDelete', async function(next) {
+  const id = this.getQuery()['_id'];
```

Comment:
It is not the universal solution because the query can be without `_id` so it is much safer to pass the whole query to the `findById` call. If this query is wrong then it is not our issue, it is an issue of business logic that calls model with wrong query

Reply:
fixed